transport/wseb/src/main/java/org/kaazing/gateway/transport/wseb/WsebConnector.java

```diff
@@ -294,12 +323,11 @@ public void initializeSession(IoSession session, T future) {
                     }
                 }
 
-                List<String> wsExtensions = connectAddressNext.getOption(WsResourceAddress.EXTENSIONS);
-                if (wsExtensions!= null) {
-                    for (String extension : wsExtensions) {
-                        if (extension != null) {
-                            httpSession.addWriteHeader(HEADER_X_WEBSOCKET_EXTENSIONS, extension);
-                        }
+                List<WebSocketExtension> extensions = webSocketExtensionFactory.offerWebSocketExtensions(
+                        (WsResourceAddress) connectAddressNext, extensionHelper);
+                for (WebSocketExtension extension : extensions) {
+                    if (extension != null) {
```

Comment:
is `if (extension != null)` better than `assert extension != null`?.  My reading of extension factory is that it can never be null